batch-stark/src/prover.rs

```diff
@@ -142,13 +272,35 @@ where
             ext_trace_domains[i].create_disjoint_domain(1 << (log_ext_degrees[i] + lqd));
 
         // Count constraints to size alpha powers packing.
-        let constraint_cnt =
-            get_symbolic_constraints(airs[i], preprocessed_widths[i], pub_vals[i].len()).len();
+        let (base_constraints, extension_constraints) = get_symbolic_constraints(
+            airs[i],
+            preprocessed_widths[i],
+            pub_vals[i].len(),
+            &all_lookups[i],
+            &lookup_data[i],
+            lookup_gadget,
+        );
+        let constraint_cnt = base_constraints.len() + extension_constraints.len();
 
         // Get evaluations on quotient domain from the main commitment.
         let trace_on_quotient_domain =
             pcs.get_evaluations_on_domain(&main_data, i, quotient_domain);
 
+        let permutation_on_quotient_domain = if let Some((_, perm_data)) =
+            &permutation_commit_and_data
+        {
+            if all_lookups[i].is_empty() {
+                None
+            } else {
+                let res =
+                    Some(pcs.get_evaluations_on_domain(perm_data, perm_counter, quotient_domain));
+                perm_counter += 1;
+                res
+            }
+        } else {
+            None
+        };
```

Comment:
Maybe something like this is clearer with less nested conditions?

```suggestion
        let permutation_on_quotient_domain = permutation_commit_and_data
        .as_ref()
        .filter(|_| !all_lookups[i].is_empty())
        .map(|(_, perm_data)| {
            let evals = pcs.get_evaluations_on_domain(perm_data, perm_counter, quotient_domain);
            perm_counter += 1;
            evals
        });
```